exercises/nucleotide-codons/src/lib.rs

```diff
@@ -1 +1,46 @@
+use std::collections::HashMap;
 
+pub struct CodonInfo<'a> {
+    actual_codons: HashMap<&'a str, &'a str>,
+}
+
+pub fn parse<'a>(pairs: Vec<(&'a str, &'a str)>) -> CodonInfo<'a> {
+    CodonInfo {
+        actual_codons: pairs.into_iter().collect(),
+    }
+}
+
+impl<'a> CodonInfo<'a> {
+    pub fn name_for(&self, codon: &str) -> Result<&'a str, &'static str> {
+        if codon.len() != 3 {
```

Comment:
> although it does not hurt to have it since it's deprecated, this lib.rs doesn't necessarily need to have a complete implementation (and we would want it not to have one if it were non-deprecated)

same